scripts/installers/make-exe

```diff
@@ -103,9 +104,24 @@ def main():
             'useful for debugging.'
         ),
     )
+    parser.add_argument(
+        '--src-dir',
+        default=None,
+        help=(
+            'Source directory for pinned installation dependencies. This '
+            'provides the ability to do offline builds without PyPI.'
+        ),
+    )
     args = parser.parse_args()
 
     output = os.path.abspath(args.output)
+    if args.src_dir:
+        print('Installing dependencies from local directory: %s' % args.src_dir)
+        install_packages(args.src_dir)
+    else:
+        run('pip install -r requirements.txt')
+        run('pip install .')
```

Comment:
Does this not need something like `pip_script` as well?

Reply:
The distinction here is one is done in a required active virtualenv and the default remains unchanged using whatever `pip` is available on the system path. We can change the `else` block to use the `pip_script` virtualenv as well, but then that means `make-exe` can't be run outside of one anymore.